Traceback (most recent call last):
  File "C:\Users\trabi\anaconda3\lib\site-packages\jupyter_cache\executors\utils.py", line 58, in single_nb_execution
    executenb(
  File "C:\Users\trabi\anaconda3\lib\site-packages\nbclient\client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "C:\Users\trabi\anaconda3\lib\site-packages\nbclient\util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "C:\Users\trabi\anaconda3\lib\site-packages\nbclient\util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "C:\Users\trabi\anaconda3\lib\asyncio\base_events.py", line 649, in run_until_complete
    return future.result()
  File "C:\Users\trabi\anaconda3\lib\site-packages\nbclient\client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "C:\Users\trabi\anaconda3\lib\site-packages\nbclient\client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "C:\Users\trabi\anaconda3\lib\site-packages\nbclient\client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# on crÃ©e une figure
plt.figure()
# on commence par un sous-graphique
plt.subplot(2,2,1)
plt.plot(np.random.randn(100))
# on enchaine avec un second sous-graphique
plt.subplot(2,2,2)
plt.plot(np.random.random(size=100),"+")
# on ajoute un troisiÃ¨me sous-graphique
plt.subplot(2,2,3)
plt.plot(np.random.randn(100),"r.")
# on finit avec le dernier sous-graphique
plt.subplot(2,2,4)
plt.plot(np.random.randn(100), "g--")
# on sauvegarde notre graphique dans un fichier
plt.savefig("../data/mes_graphiques.jpg")
#plt.show() #si dans un IDE classique
------------------

[1;31m---------------------------------------------------------------------------[0m
[1;31mFileNotFoundError[0m                         Traceback (most recent call last)
Cell [1;32mIn[4], line 16[0m
[0;32m     14[0m plt[38;5;241m.[39mplot(np[38;5;241m.[39mrandom[38;5;241m.[39mrandn([38;5;241m100[39m), [38;5;124m"[39m[38;5;124mg--[39m[38;5;124m"[39m)
[0;32m     15[0m [38;5;66;03m# on sauvegarde notre graphique dans un fichier[39;00m
[1;32m---> 16[0m [43mplt[49m[38;5;241;43m.[39;49m[43msavefig[49m[43m([49m[38;5;124;43m"[39;49m[38;5;124;43m../data/mes_graphiques.jpg[39;49m[38;5;124;43m"[39;49m[43m)[49m

File [1;32m~\anaconda3\lib\site-packages\matplotlib\pyplot.py:996[0m, in [0;36msavefig[1;34m(*args, **kwargs)[0m
[0;32m    993[0m [38;5;129m@_copy_docstring_and_deprecators[39m(Figure[38;5;241m.[39msavefig)
[0;32m    994[0m [38;5;28;01mdef[39;00m [38;5;21msavefig[39m([38;5;241m*[39margs, [38;5;241m*[39m[38;5;241m*[39mkwargs):
[0;32m    995[0m     fig [38;5;241m=[39m gcf()
[1;32m--> 996[0m     res [38;5;241m=[39m fig[38;5;241m.[39msavefig([38;5;241m*[39margs, [38;5;241m*[39m[38;5;241m*[39mkwargs)
[0;32m    997[0m     fig[38;5;241m.[39mcanvas[38;5;241m.[39mdraw_idle()  [38;5;66;03m# Need this if 'transparent=True', to reset colors.[39;00m
[0;32m    998[0m     [38;5;28;01mreturn[39;00m res

File [1;32m~\anaconda3\lib\site-packages\matplotlib\figure.py:3328[0m, in [0;36mFigure.savefig[1;34m(self, fname, transparent, **kwargs)[0m
[0;32m   3324[0m     [38;5;28;01mfor[39;00m ax [38;5;129;01min[39;00m [38;5;28mself[39m[38;5;241m.[39maxes:
[0;32m   3325[0m         stack[38;5;241m.[39menter_context(
[0;32m   3326[0m             ax[38;5;241m.[39mpatch[38;5;241m.[39m_cm_set(facecolor[38;5;241m=[39m[38;5;124m'[39m[38;5;124mnone[39m[38;5;124m'[39m, edgecolor[38;5;241m=[39m[38;5;124m'[39m[38;5;124mnone[39m[38;5;124m'[39m))
[1;32m-> 3328[0m [38;5;28mself[39m[38;5;241m.[39mcanvas[38;5;241m.[39mprint_figure(fname, [38;5;241m*[39m[38;5;241m*[39mkwargs)

File [1;32m~\anaconda3\lib\site-packages\matplotlib\backend_bases.py:2362[0m, in [0;36mFigureCanvasBase.print_figure[1;34m(self, filename, dpi, facecolor, edgecolor, orientation, format, bbox_inches, pad_inches, bbox_extra_artists, backend, **kwargs)[0m
[0;32m   2358[0m [38;5;28;01mtry[39;00m:
[0;32m   2359[0m     [38;5;66;03m# _get_renderer may change the figure dpi (as vector formats[39;00m
[0;32m   2360[0m     [38;5;66;03m# force the figure dpi to 72), so we need to set it again here.[39;00m
[0;32m   2361[0m     [38;5;28;01mwith[39;00m cbook[38;5;241m.[39m_setattr_cm([38;5;28mself[39m[38;5;241m.[39mfigure, dpi[38;5;241m=[39mdpi):
[1;32m-> 2362[0m         result [38;5;241m=[39m print_method(
[0;32m   2363[0m             filename,
[0;32m   2364[0m             facecolor[38;5;241m=[39mfacecolor,
[0;32m   2365[0m             edgecolor[38;5;241m=[39medgecolor,
[0;32m   2366[0m             orientation[38;5;241m=[39morientation,
[0;32m   2367[0m             bbox_inches_restore[38;5;241m=[39m_bbox_inches_restore,
[0;32m   2368[0m             [38;5;241m*[39m[38;5;241m*[39mkwargs)
[0;32m   2369[0m [38;5;28;01mfinally[39;00m:
[0;32m   2370[0m     [38;5;28;01mif[39;00m bbox_inches [38;5;129;01mand[39;00m restore_bbox:

File [1;32m~\anaconda3\lib\site-packages\matplotlib\backend_bases.py:2228[0m, in [0;36mFigureCanvasBase._switch_canvas_and_return_print_method.<locals>.<lambda>[1;34m(*args, **kwargs)[0m
[0;32m   2224[0m     optional_kws [38;5;241m=[39m {  [38;5;66;03m# Passed by print_figure for other renderers.[39;00m
[0;32m   2225[0m         [38;5;124m"[39m[38;5;124mdpi[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124mfacecolor[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124medgecolor[39m[38;5;124m"[39m, [38;5;124m"[39m[38;5;124morientation[39m[38;5;124m"[39m,
[0;32m   2226[0m         [38;5;124m"[39m[38;5;124mbbox_inches_restore[39m[38;5;124m"[39m}
[0;32m   2227[0m     skip [38;5;241m=[39m optional_kws [38;5;241m-[39m {[38;5;241m*[39minspect[38;5;241m.[39msignature(meth)[38;5;241m.[39mparameters}
[1;32m-> 2228[0m     print_method [38;5;241m=[39m functools[38;5;241m.[39mwraps(meth)([38;5;28;01mlambda[39;00m [38;5;241m*[39margs, [38;5;241m*[39m[38;5;241m*[39mkwargs: meth(
[0;32m   2229[0m         [38;5;241m*[39margs, [38;5;241m*[39m[38;5;241m*[39m{k: v [38;5;28;01mfor[39;00m k, v [38;5;129;01min[39;00m kwargs[38;5;241m.[39mitems() [38;5;28;01mif[39;00m k [38;5;129;01mnot[39;00m [38;5;129;01min[39;00m skip}))
[0;32m   2230[0m [38;5;28;01melse[39;00m:  [38;5;66;03m# Let third-parties do as they see fit.[39;00m
[0;32m   2231[0m     print_method [38;5;241m=[39m meth

File [1;32m~\anaconda3\lib\site-packages\matplotlib\backends\backend_agg.py:526[0m, in [0;36mFigureCanvasAgg.print_jpg[1;34m(self, filename_or_obj, pil_kwargs)[0m
[0;32m    521[0m [38;5;28;01mdef[39;00m [38;5;21mprint_jpg[39m([38;5;28mself[39m, filename_or_obj, [38;5;241m*[39m, pil_kwargs[38;5;241m=[39m[38;5;28;01mNone[39;00m):
[0;32m    522[0m     [38;5;66;03m# savefig() has already applied savefig.facecolor; we now set it to[39;00m
[0;32m    523[0m     [38;5;66;03m# white to make imsave() blend semi-transparent figures against an[39;00m
[0;32m    524[0m     [38;5;66;03m# assumed white background.[39;00m
[0;32m    525[0m     [38;5;28;01mwith[39;00m mpl[38;5;241m.[39mrc_context({[38;5;124m"[39m[38;5;124msavefig.facecolor[39m[38;5;124m"[39m: [38;5;124m"[39m[38;5;124mwhite[39m[38;5;124m"[39m}):
[1;32m--> 526[0m         [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_print_pil[49m[43m([49m[43mfilename_or_obj[49m[43m,[49m[43m [49m[38;5;124;43m"[39;49m[38;5;124;43mjpeg[39;49m[38;5;124;43m"[39;49m[43m,[49m[43m [49m[43mpil_kwargs[49m[43m)[49m

File [1;32m~\anaconda3\lib\site-packages\matplotlib\backends\backend_agg.py:458[0m, in [0;36mFigureCanvasAgg._print_pil[1;34m(self, filename_or_obj, fmt, pil_kwargs, metadata)[0m
[0;32m    453[0m [38;5;124;03m"""[39;00m
[0;32m    454[0m [38;5;124;03mDraw the canvas, then save it using `.image.imsave` (to which[39;00m
[0;32m    455[0m [38;5;124;03m*pil_kwargs* and *metadata* are forwarded).[39;00m
[0;32m    456[0m [38;5;124;03m"""[39;00m
[0;32m    457[0m FigureCanvasAgg[38;5;241m.[39mdraw([38;5;28mself[39m)
[1;32m--> 458[0m [43mmpl[49m[38;5;241;43m.[39;49m[43mimage[49m[38;5;241;43m.[39;49m[43mimsave[49m[43m([49m
[0;32m    459[0m [43m    [49m[43mfilename_or_obj[49m[43m,[49m[43m [49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mbuffer_rgba[49m[43m([49m[43m)[49m[43m,[49m[43m [49m[38;5;28;43mformat[39;49m[38;5;241;43m=[39;49m[43mfmt[49m[43m,[49m[43m [49m[43morigin[49m[38;5;241;43m=[39;49m[38;5;124;43m"[39;49m[38;5;124;43mupper[39;49m[38;5;124;43m"[39;49m[43m,[49m
[0;32m    460[0m [43m    [49m[43mdpi[49m[38;5;241;43m=[39;49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mfigure[49m[38;5;241;43m.[39;49m[43mdpi[49m[43m,[49m[43m [49m[43mmetadata[49m[38;5;241;43m=[39;49m[43mmetadata[49m[43m,[49m[43m [49m[43mpil_kwargs[49m[38;5;241;43m=[39;49m[43mpil_kwargs[49m[43m)[49m

File [1;32m~\anaconda3\lib\site-packages\matplotlib\image.py:1687[0m, in [0;36mimsave[1;34m(fname, arr, vmin, vmax, cmap, format, origin, dpi, metadata, pil_kwargs)[0m
[0;32m   1685[0m pil_kwargs[38;5;241m.[39msetdefault([38;5;124m"[39m[38;5;124mformat[39m[38;5;124m"[39m, [38;5;28mformat[39m)
[0;32m   1686[0m pil_kwargs[38;5;241m.[39msetdefault([38;5;124m"[39m[38;5;124mdpi[39m[38;5;124m"[39m, (dpi, dpi))
[1;32m-> 1687[0m image[38;5;241m.[39msave(fname, [38;5;241m*[39m[38;5;241m*[39mpil_kwargs)

File [1;32m~\anaconda3\lib\site-packages\PIL\Image.py:2410[0m, in [0;36mImage.save[1;34m(self, fp, format, **params)[0m
[0;32m   2408[0m         fp [38;5;241m=[39m builtins[38;5;241m.[39mopen(filename, [38;5;124m"[39m[38;5;124mr+b[39m[38;5;124m"[39m)
[0;32m   2409[0m     [38;5;28;01melse[39;00m:
[1;32m-> 2410[0m         fp [38;5;241m=[39m [43mbuiltins[49m[38;5;241;43m.[39;49m[43mopen[49m[43m([49m[43mfilename[49m[43m,[49m[43m [49m[38;5;124;43m"[39;49m[38;5;124;43mw+b[39;49m[38;5;124;43m"[39;49m[43m)[49m
[0;32m   2412[0m [38;5;28;01mtry[39;00m:
[0;32m   2413[0m     save_handler([38;5;28mself[39m, fp, filename)

[1;31mFileNotFoundError[0m: [Errno 2] No such file or directory: '../data/mes_graphiques.jpg'
FileNotFoundError: [Errno 2] No such file or directory: '../data/mes_graphiques.jpg'

